### PR TITLE
Update BUILDING-OPENSUSE.md

### PR DESCRIPTION
**Pull request by saif-ellafi**
Build from source in OpenSUSE Tumbleweed - tested

Best regards!

### CODE REVIEW

File: docs/BUILDING-OPENSUSE.md [hunk 1]
```diff
@@ -19,11 +19,12 @@ sudo zypper in obs-studio
...
 sudo zypper install zsh cmake Mesa-libGL-devel \
-  ffmpeg-6-libavcodes-devel ffmpeg-6-libavdevice-devel ffmpeg-6-libavformat-devel \
+  ffmpeg-6-libavcodec-devel ffmpeg-6-libavdevice-devel ffmpeg-6-libavformat-devel \
   libcurl-devel Mesa-libEGL-devel \
   libpulse-devel libxkbcommon-devel
-</dev/null >.github/scripts/utils.zsh/check_linux
-CI=1 .github/scripts/build-linux.zsh --skip-deps
+sudo zypper in cmake gcc-c++ ninja obs-studio-devel opencv-devel qt6-base-devel zsh curl-devel jq
+
+.github/scripts/build-linux --skip-deps
```
Comment (umireon) [inline docs/BUILDING-OPENSUSE.md]:
```suggestion
cmake . -B build_x86_64 \
  -DCMAKE_C_COMPILER=gcc-12 \
  -DCMAKE_CXX_COMPILER=g++-12 \
  -DQT_VERSION=6 \
  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
  -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
  -DENABLE_FRONTEND_API=ON \
  -DENABLE_QT=ON
```

Can you test this?

**Review by umireon — CHANGES_REQUESTED** [review]

**Comment by umireon** [comment]
Can you remove 

```
sudo zypper install zsh cmake Mesa-libGL-devel \
  ffmpeg-6-libavcodes-devel ffmpeg-6-libavdevice-devel ffmpeg-6-libavformat-devel \
  ffmpeg-6-libavcodec-devel ffmpeg-6-libavdevice-devel ffmpeg-6-libavformat-devel \
  libcurl-devel Mesa-libEGL-devel \
  libpulse-devel libxkbcommon-devel
```

**Comment by umireon** [comment]
@saif-ellafi Hello can you update the branch as I requested?

**Comment by saif-ellafi** [comment]
Very sorry @umireon - I moved out of opensuse, can't test. But I can say for sure the script as committed worked fine (CPU only)

**Comment by saif-ellafi** [comment]
@umireon - will test soon and come back to you

**Comment by saif-ellafi** [comment]
@umireon - doesn't work

```
saif@saif-ms7c76:~/Repos/obs-backgroundremoval> cmake . -B build_x86_64   -DCMAKE_C_COMPILER=gcc-12   -DCMAKE_CXX_COMPILER=g++-12   -DQT_VERSION=6   -DCMAKE_BUILD_TYPE=RelWithDebInfo   -DCMAKE_POSITION_INDEPENDENT_CODE=ON   -DENABLE_FRONTEND_API=ON   -DENABLE_QT=ON
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:5 (project):
  The CMAKE_C_COMPILER:

    gcc-12

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


CMake Error at CMakeLists.txt:5 (project):
  The CMAKE_CXX_COMPILER:

    g++-12

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
```

**Comment by saif-ellafi** [comment]
Bumped to g++ 13 but now getting

```
CMake Error at CMakeLists.txt:14 (add_library):
  The install of the obs-backgroundremoval target requires changing an RPATH
  from the build tree, but this is not supported with the Ninja generator
  unless on an ELF-based or XCOFF-based platform.  The
  CMAKE_BUILD_WITH_INSTALL_RPATH variable may be set to avoid this relinking
  step.
```

File: docs/BUILDING-OPENSUSE.md [hunk 2]
```diff
@@ -19,11 +19,19 @@ sudo zypper in obs-studio
 ```
 sudo zypper install -t pattern devel_basis
 sudo zypper install zsh cmake Mesa-libGL-devel \
-  ffmpeg-6-libavcodes-devel ffmpeg-6-libavdevice-devel ffmpeg-6-libavformat-devel \
+  ffmpeg-6-libavcodec-devel ffmpeg-6-libavdevice-devel ffmpeg-6-libavformat-devel \
   libcurl-devel Mesa-libEGL-devel \
   libpulse-devel libxkbcommon-devel
-</dev/null >.github/scripts/utils.zsh/check_linux
-CI=1 .github/scripts/build-linux.zsh --skip-deps
+sudo zypper in cmake gcc-c++ ninja obs-studio-devel opencv-devel qt6-base-devel zsh curl-devel jq
```
Comment (umireon) [inline docs/BUILDING-OPENSUSE.md]:
```suggestion
sudo zypper in cmake gcc12-c++ ninja obs-studio-devel opencv-devel qt6-base-devel zsh curl-devel jq
```

**Review by umireon — CHANGES_REQUESTED** [review]

**Comment by umireon** [comment]
Please use gcc12 not gcc13 as gcc13 is not avaible on Leap

**Comment by umireon** [comment]
You should not use .github/scripts/build-linux.zsh

**Comment by umireon** [comment]
You have to remove build_x86_64 directory before running cmake command

File: docs/BUILDING-OPENSUSE.md [hunk 3]
```diff
@@ -19,11 +19,19 @@ sudo zypper in obs-studio
...
+  ffmpeg-6-libavcodec-devel ffmpeg-6-libavdevice-devel ffmpeg-6-libavformat-devel \
   libcurl-devel Mesa-libEGL-devel \
   libpulse-devel libxkbcommon-devel
-</dev/null >.github/scripts/utils.zsh/check_linux
-CI=1 .github/scripts/build-linux.zsh --skip-deps
+sudo zypper in cmake gcc12-c++ ninja obs-studio-devel opencv-devel qt6-base-devel zsh curl-devel jq
+
+cmake . -B build_x86_64 \
+  -DCMAKE_C_COMPILER=gcc-13 \
+  -DCMAKE_CXX_COMPILER=g++-13 \
```
Comment (umireon) [inline docs/BUILDING-OPENSUSE.md]:
```suggestion
  -DCMAKE_C_COMPILER=gcc-12 \
  -DCMAKE_CXX_COMPILER=g++-12 \
```

**Review by umireon — CHANGES_REQUESTED** [review]

**Comment by saif-ellafi** [comment]
Okay cmake works but the install is failing

```
sudo cmake --install build_x86_64 --prefix /usr
-- Install configuration: "RelWithDebInfo"
-- Installing: /usr/lib64/obs-plugins/obs-backgroundremoval/libonnxruntime.so.1.16.0
-- Installing: /usr/lib64/obs-plugins/obs-backgroundremoval/libonnxruntime_providers_shared.so
-- Installing: /usr/lib64/obs-plugins/obs-backgroundremoval/libonnxruntime_providers_cuda.so
-- Installing: /usr/lib64/obs-plugins/obs-backgroundremoval/libonnxruntime_providers_tensorrt.so
CMake Error at build_x86_64/cmake_install.cmake:66 (file):
  file INSTALL cannot find
  "/home/saif/Repos/obs-backgroundremoval/build_x86_64/obs-backgroundremoval.so":
  No such file or directory.
```

File: docs/BUILDING-OPENSUSE.md [hunk 4]
```diff
@@ -19,11 +19,19 @@ sudo zypper in obs-studio
...
+sudo zypper in cmake gcc12-c++ ninja obs-studio-devel opencv-devel qt6-base-devel zsh curl-devel jq
+
+cmake . -B build_x86_64 \
+  -DCMAKE_C_COMPILER=gcc-12 \
+  -DCMAKE_CXX_COMPILER=g++-12 \
+  -DQT_VERSION=6 \
+  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+  -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+  -DENABLE_FRONTEND_API=ON \
+  -DENABLE_QT=ON
```
Comment (umireon) [inline docs/BUILDING-OPENSUSE.md:34]:
```suggestion
  -DENABLE_QT=ON
cmake --build build_x86_64
```

**Review by umireon — CHANGES_REQUESTED** [review]

**Comment by umireon** [comment]
Run cmake --build build_x86_64 before installing.

**Comment by saif-ellafi** [comment]
@umireon - thanks. Now it seems to build and install properly. CPU only. Will check now to install cuda and the other mambo-jumbo for GPU support

**Review by umireon — APPROVED** [review]
LGTM

**Pull request merged**